library/init/meta/interactive.lean

```diff
@@ -791,8 +791,14 @@ tactic.reflexivity
 meta def symmetry : tactic unit :=
 tactic.symmetry
 
-meta def transitivity : tactic unit :=
-tactic.transitivity
+meta def transitivity (q : parse texpr?) : tactic unit :=
+tactic.transitivity >> match q with
+| none := skip
+| some q := do t ← target, match t with
+  | (app (app r lhs) rhs) := i_to_expr q >>= unify rhs
```

Comment:
This does not work with `heq {α : Sort u} (a : α) : Π {β : Sort u}, β → Pro` where the arguments are the second and last parameter. I guess you want to use `relation_lhs_rhs` to project out the lhs and rhs side.